esphome/components/modbus_controller/switch/__init__.py

```diff
@@ -62,7 +63,9 @@ async def to_code(config):
     paren = await cg.get_variable(config[CONF_MODBUS_CONTROLLER_ID])
     cg.add(var.set_parent(paren))
     cg.add(var.set_use_write_mutiple(config[CONF_USE_WRITE_MULTIPLE]))
-    cg.add(paren.add_sensor_item(var))
+    cg.add(var.set_assumed_state(config[CONF_ASSUMED_STATE]))
+    if config[CONF_ASSUMED_STATE] is False:
```

Comment:
If something is used more than once it should be a variable
```suggestion
    assumed_state = config[CONF_ASSUMED_STATE]
    cg.add(var.set_assumed_state(assumed_state))
    if not assumed_state:
```